be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -699,6 +699,11 @@ Status SegmentIterator::_get_row_ranges_from_conditions(RowRanges* condition_row
         if (_opts.io_ctx.reader_type == ReaderType::READER_QUERY) {
             RowRanges dict_row_ranges = RowRanges::create_single(num_rows());
             for (auto cid : cids) {
+                if (!_segment->can_apply_predicate_safely(cid,
```

Comment:
There is check for can_apply_predicate_safely() in bf and zonemap branch. Why it's not check in this branch before? Does this modification affect other logic in this branch?

Reply:
only takes effect when storage type and predicate type ares missmatched typically in variant predicates with cast